src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs

```diff
@@ -223,6 +237,11 @@ public void AddFiles(string filePaths)
             }
         }
 
+        private bool isSupportedDeviceType(Edition engineEdition, DeviceType deviceType)
+        {
+            return supportedDeviceTypes.ContainsKey(engineEdition) && supportedDeviceTypes[engineEdition].Contains(deviceType);
```

Comment:
I'd rather we not require that the edition be known since that would mean that if a new edition is added later on we'd have to remember to go update this to add it to the dictionary. 

Instead we should have a "DefaultSupportedDeviceTypes" - and then the dictionary would only contain "special" editions such as MI.

Reply:
Is this better?